### PR TITLE
fix: respect ARGOCD_GIT_MODULES_ENABLED in the appset controller (#10285)

### LINKED ISSUE #10285: ApplicationSet no longer honoring ARGOCD_GIT_MODULES_ENABLED env var
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

After upgrading ArgoCD from 2.2.3 to 2.4.8 and removing the extra applicationset helm chart dependency, the applicationset controller no longer seems to respect the `ARGOCD_GIT_MODULES_ENABLED: "false"`  env var, and cannot generate certain applications.

**To Reproduce**

Deploy ArgoCD helm chart v 2.4.8 with
```yaml
  applicationSet:
    enabled: true
    extraEnv:
      - name: "ARGOCD_GIT_MODULES_ENABLED"
        value: "false"
```

**Expected behavior**

I expect the applicationset controller to ignore submodules when I set the env var `ARGOCD_GIT_MODULES_ENABLED` to "false"

**Screenshots**

**Version**

I am not using cli, only using the UI.
```
v2.4.8+844f79e
```

**Logs**

```
time="2022-08-10T21:07:12Z" level=info msg=Trace args="[git submodule sync --recursive]" dir=/tmp/git@github.com_***_*** operation_name="exec git" time_ms=14.129612999999999

time="2022-08-10T21:07:12Z" level=info msg="git submodule update --init --recursive" dir=/tmp/git@github.com_***_*** execID=d250a

time="2022-08-10T21:07:12Z" level=error msg="`git submodule update --init --recursive` failed exit status 1: Cloning into '/tmp/git@github.com_***_***/***'...\nERROR: Repository not found.\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\nfatal: clone of 'git@github.com:***/***.git' into submodule path '/tmp/git@github.com_***_***' failed\nFailed to clone '***'. Retry scheduled\nCloning into '/tmp/git@github.com_***/***'...\nERROR: Repository not found.\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\nfatal: clone of 'git@github.com:***/***.git' into submodule path '/tmp/git@github.com_***_***' failed\nFailed to clone '***' a second time, aborting" execID=d250a

```

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Fixes #10285

### CODE REVIEW

File: cmd/argocd-applicationset-controller/commands/applicationset_controller.go [hunk 1]
```diff
@@ -210,6 +216,7 @@ func NewCommand() *cobra.Command {
 	command.Flags().StringVar(&logLevel, "loglevel", "info", "Set the logging level. One of: debug|info|warn|error")
 	command.Flags().BoolVar(&dryRun, "dry-run", false, "Enable dry run mode")
 	command.Flags().StringVar(&logFormat, "logformat", "text", "Set the logging format. One of: text|json")
+	command.Flags().StringVar(&logFormat, "logformat", "text", "Set the logging format. One of: text|json")
```
Comment (notfromstatefarm) [inline cmd/argocd-applicationset-controller/commands/applicationset_controller.go:219]: doubled up on this line somehow?

**Review by notfromstatefarm — CHANGES_REQUESTED** [review]
LGTM other than doubled up `StringVar`

File: cmd/argocd-applicationset-controller/commands/applicationset_controller.go (hunk 1 above)
Comment (crenshaw-dev) [reply cmd/argocd-applicationset-controller/commands/applicationset_controller.go]: Oops! Fixed

**Review by alexmt — APPROVED** [review]
Looks good!

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.4 for 2.4.10.

**Pull request merged**